packages/frontend/src/pages/Executions/index.tsx

```diff
@@ -1,45 +1,84 @@
 import type { IExecution } from '@plumber/types'
 
 import * as React from 'react'
+import { BiSearch } from 'react-icons/bi'
 import { Link, useSearchParams } from 'react-router-dom'
 import { useQuery } from '@apollo/client'
+import {
+  Divider as ChakraDivider,
+  Flex,
+  Icon,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  InputRightElement,
+} from '@chakra-ui/react'
 import Box from '@mui/material/Box'
 import CircularProgress from '@mui/material/CircularProgress'
 import Divider from '@mui/material/Divider'
-import Grid from '@mui/material/Grid'
 import Pagination from '@mui/material/Pagination'
 import PaginationItem from '@mui/material/PaginationItem'
 import Container from 'components/Container'
 import ExecutionRow from 'components/ExecutionRow'
+import ExecutionStatusMenu from 'components/ExecutionStatusMenu'
 import NoResultFound from 'components/NoResultFound'
 import PageTitle from 'components/PageTitle'
 import { GET_EXECUTIONS } from 'graphql/queries/get-executions'
 import useFormatMessage from 'hooks/useFormatMessage'
+import debounce from 'lodash/debounce'
 
 const EXECUTION_PER_PAGE = 10
 const EXECUTIONS_TITLE = 'Executions'
 
-const getLimitAndOffset = (page: number) => ({
+const getLimitAndOffset = (
+  page: number,
+  filterStatus: string,
+  searchInput: string,
+) => ({
   limit: EXECUTION_PER_PAGE,
   offset: (page - 1) * EXECUTION_PER_PAGE,
+  status: filterStatus,
+  searchInput: searchInput,
 })
 
 export default function Executions(): React.ReactElement {
   const formatMessage = useFormatMessage()
   const [searchParams, setSearchParams] = useSearchParams()
   const page = parseInt(searchParams.get('page') || '', 10) || 1
 
+  const [searchInput, setSearchInput] = React.useState<string>('')
+
+  const handleSearchValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(event.target.value)
+    setSearchParams({ status: filterStatus })
+  }
+  const handleSearchValueDebounced = debounce(handleSearchValue, 1000)
+
+  const [filterStatus, setFilterStatus] = React.useState<string>(
+    searchParams.get('status') || '',
+  )
+
+  const onFilterChange = (status: string) => {
+    setFilterStatus(status)
+    setSearchParams({ status })
+  }
+
+  // checks for updates to re-render execution status menu
+  React.useEffect(() => {
+    setFilterStatus(searchParams.get('status') || '')
+  }, [searchParams])
+
   const { data, refetch, loading } = useQuery(GET_EXECUTIONS, {
-    variables: getLimitAndOffset(page),
+    variables: getLimitAndOffset(page, filterStatus, searchInput),
     fetchPolicy: 'cache-and-network',
     pollInterval: 5000,
   })
   const getExecutions = data?.getExecutions || {}
   const { pageInfo, edges } = getExecutions
 
   React.useEffect(() => {
-    refetch(getLimitAndOffset(page))
-  }, [refetch, page])
+    refetch(getLimitAndOffset(page, filterStatus.toLowerCase(), searchInput))
```

Comment:
i know this was useEffect was there previously, but i think it auto refetches when the variables change cos it is stored in a useState hook, alr causing a reflow